.github/workflows/ci.yaml

```diff
@@ -258,14 +260,14 @@ jobs:
       id: fetch_sdist
       uses: actions/download-artifact@v4
       with:
-        name: ${{ needs.build_sdist.outputs.artifact_name }}
```

Comment:
*(this was an empty string because there's no job with ID `build_sdist`; causing all the artifacts to be downloaded and unpacked in subdirectories)*

Reply:
Ugh- another bug introduced by moving back and forth between PyYAML and CFFI trying to standardize the workflows between them. GHA doesn't make it easy to debug these kinds of things because so many different layers just silently succeed with glaring issues like this. Their (lack of) expression type system also makes me appreciate Jinja *so* much more.

Reply:
Interestingly, in the past, their YAML parser would error out if you were to reference an identifier not listed in the `needs:` key. It used to be rather hard to find that web page that displays the error too since the checks wouldn't appear in the PR.